libsql-sqlite3/src/main.c

```diff
@@ -3352,14 +3339,23 @@ static int openDatabase(
   /* Allocate the sqlite data structure */
   db = sqlite3MallocZero( sizeof(sqlite3) );
   if( db==0 ) goto opendb_out;
-  db->pWalMethodsData = pWalMethodsData;
+  rc = make_ref_counted_create_wal(create_wal, &(db->create_wal));
```

Comment:
That adds allocation overhead to the regular path with regular WAL. Let's create a static "ref counted" instance with refcount like i32::max or something, and use it in the regular case, when there's no custom WAL?